pex/finders.py

```diff
@@ -300,15 +328,22 @@ def get_entrypoint(dist):
     script_entry = dist.get_entry_map().get('console_scripts', {}).get(script)
     if script_entry is not None:
       # Entry points are of the form 'foo = bar', we just want the 'bar' part.
-      return dist.key, str(script_entry).split('=')[1].strip()
+      return str(script_entry).split('=')[1].strip()
 
-  entries = frozenset(filter(None, (get_entrypoint(dist) for dist in dists)))
+  entries = {}
```

Comment:
Was going to suggest a dictionary comprehension but I think I see why you’re using a loop.

This is why Python 3.8 will introduce the assignment operator, so that you could use a dictionary comprehension without evaluating `get_entrypoint(dist)` twice

Reply:
Nice, I did not know about PEP-572. Yeah, this is exactly why I had to break out a loop.